yarn-project/aztec/src/cli/cmds/start_pxe.ts

```diff
@@ -30,6 +72,41 @@ export async function addPXE(
   const pxe = await createPXEService(node, pxeConfig);
   const pxeServer = createPXERpcServer(pxe);
 
+  // register basic contracts
+  if (pxeCliOptions.network) {
+    userLog(`Registering basic contracts for ${pxeCliOptions.network}`);
+    const basicContractsInfo = await fetchBasicContractAddresses(
+      `${contractAddressesUrl}/${pxeCliOptions.network}/basic_contracts.json`,
+    );
+    const l2Contracts: Record<
+      string,
+      { name: string; address: AztecAddress; initHash: Fr; salt: Fr; artifact: ContractArtifact }
+    > = {};
+    for (const [key, artifactName] of Object.entries(L2BasicContractsMap[pxeCliOptions.network as Network])) {
+      l2Contracts[key] = {
+        name: key,
+        address: AztecAddress.fromString(basicContractsInfo[`${key}L2`]),
+        initHash: Fr.fromString(basicContractsInfo[`${key}L2InitHash`]),
+        salt: Fr.fromString(basicContractsInfo[`${key}L2Salt`]),
+        artifact: await getContractArtifact(artifactName, userLog),
```

Comment:
Might be worth refactoring how the contracts are saved on static.aztec.network now, before we have to maintain backwards compatibility, so that in the future if we do add new contracts, this code won't need to change. I'm thinking we could save them like this
```
{
  "testCounter": {
    "address": "...",
    "initializationHash": "....",
    "salt": "....",
    "deployerAddress": "..."
  }
}
```

Then loading them up in the pxe becomes just an iteration over the top-level keys.

Reply:
This isn't something that has to be done in this PR but we should consider doing it soon :tm:

Reply:
+1